src/api/VoteMonitor.Api.Note/Controllers/NoteController.cs

```diff
@@ -46,30 +45,37 @@ public async Task<List<NoteModel>> Get(NoteQuery filter)
         /// <param name="note"></param>
         /// <returns></returns>
         [HttpPost("upload")]
-        public async Task<dynamic> Upload(IFormFile file, [FromForm]NoteModel note)
+        public async Task<IActionResult> Upload(List<IFormFile> files, [FromForm]NoteModel note)
         {
             if (!ModelState.IsValid)
-                return this.ResultAsync(HttpStatusCode.BadRequest);
+                return BadRequest();
 
             // TODO[DH] use a pipeline instead of separate Send commands
             // daca nota este asociata sectiei
             var idSectie = await _mediator.Send(_mapper.Map<PollingStationQuery>(note));
             if (idSectie < 0)
-                return this.ResultAsync(HttpStatusCode.NotFound);
+                return NotFound();
 
             var command = _mapper.Map<AddNoteCommand>(note);
-            var fileAddress = await _mediator.Send(new UploadFileCommand { File = file });
+            var fileAddresses = await _mediator.Send(new UploadFileCommand { Files = files });
 
-            command.IdObserver = int.Parse(User.Claims.First(c => c.Type == ClaimsHelper.ObserverIdProperty).Value);
-            command.AttachementPath = fileAddress;
+            command.IdObserver = int.Parse(User.Claims.First(c => c.Type == ClaimsHelper.ObserverIdProperty).Value); ;
+            command.AttachementPaths = fileAddresses;
             command.IdPollingStation = idSectie;
 
-            var result = await _mediator.Send(command);
-
-            if (result < 0)
-                return this.ResultAsync(HttpStatusCode.NotFound);
-
-            return await Task.FromResult(new { FileAdress = fileAddress, note });
+            try
+            {
+                var result = await _mediator.Send(command);
+                return Ok(result);
```

Comment:
I think we should use `NoteModel` as the return type here, since the `api/v2/note GET` endpoint also returns a `List<NoteModel>`

Reply:
Same as the upper comment